codecov_auth/signals.py

```diff
@@ -45,3 +48,38 @@ def update_org_token(sender, instance: OrganizationLevelToken, **kwargs):
                 }
             ).encode("utf-8"),
         )
+
+
+@receiver(post_save, sender=Owner, dispatch_uid="shelter_sync_owner")
+def update_owner(sender, instance: Owner, **kwargs):
+    """
+    Shelter tracks a limited set of Owner fields - only update if those fields have changed.
+    """
+    created = kwargs["created"]
+    tracked_fields = [
+        "upload_token_required_for_public_repos",
+        "username",
+        "service",
+    ]
+    if created or any(instance.tracker.has_changed(field) for field in tracked_fields):
+        pubsub_project_id = settings.SHELTER_PUBSUB_PROJECT_ID
```

Comment:
Optional: since this is used in a couple of places, would encourage to move this into its own fn and use that in every signal for readability!

Reply:
I agree - I added one, check it out!